common/chan_utils_test.go

```diff
@@ -0,0 +1,200 @@
+		}
+	})
+
+	t.Run("send success,Unbuffered chan", func(t *testing.T) {
+		c := make(chan string)
+		ctx := context.Background()
+		var wg sync.WaitGroup
+		wg.Add(2)
+		go func() {
+			defer wg.Done()
+			SendToChannel(ctx, c, "data")
+		}()
+		go func() {
+			defer wg.Done()
+			value := <-c
+			// Verify if the received data is correct
+			if value != "data" {
+				t.Errorf("expected to receive \"data\" from channel, but received %s", value)
+			}
+		}()
+		wg.Wait()
+	})
+
+	//context timeout
+	//The first t.Run is a test with a timeout setting, but it was successfully sent without timeout
+	//The second t.Run passes through time.Sleep setting timeout, simulating context timeout
+	t.Run("context not timeout", func(t *testing.T) {
+		c := make(chan string, 1)
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Millisecond)
+		defer cancel()
+		if SendToChannel(ctx, c, "hello") {
+			t.Log("Data sent successfully")
+		} else {
+			fmt.Println(SendToChannel(ctx, c, "hello"))
+			t.Fatal("Failed to send data due to context timeout")
+		}
+		select {
+		case <-c:
+			t.Log("Successfully received data")
+		case <-ctx.Done():
+			t.Fatal("Channel closed after context timeout")
+		}
+	})
+
+	t.Run("context timeout", func(t *testing.T) {
+		c := make(chan string, 1)
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Millisecond)
+		defer cancel()
+		if SendToChannel(ctx, c, "data") {
+			t.Log("Data sent successfully")
+		} else {
+			t.Fatal("Failed to send data due to context timeout")
+		}
+
+		go func() {
+			time.Sleep(time.Second) // Set timeout
+			select {
+			case <-c:
+				t.Error("Timed out but able to retrieve data")
+				return
+			case <-ctx.Done():
+				t.Log("No data received due to context timeout")
+			}
+		}()
+	})
+
+	//incorrect type
+	//It is not necessary to distinguish between buffered and unbuffered, as it is necessary to test the incorrect type
+	t.Run("incorrect type", func(t *testing.T) {
+		defer func() {
+			if r := recover(); r != nil {
+				t.Log("test-ok")
+			} else {
+				t.Log("test-fail")
+			}
+		}()
+		c := make(chan int)
+		ctx := context.Background()
+		SendToChannel(ctx, c, "incorrect type")
+	})
+}
+
+func TestZeroValue(t *testing.T) {
+	tests := []struct {
+		Type string
+		want interface{}
+	}{
+		{Type: "int", want: 0},
+		{Type: "string", want: ""},
+		{Type: "bool", want: false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.Type, func(t *testing.T) {
+			zero := zeroValue[interface{}]()
+			switch v := zero.(type) {
+			case int:
+				if v != 0 {
+					t.Errorf("zeroValue = %v, want 0", v)
+				}
+			case string:
+				if v != "" {
+					t.Errorf("zeroValue = %v, want \"\"", v)
+				}
+			case bool:
+				if v != false {
+					t.Errorf("zeroValue = %v, want false", v)
+				}
+			default:
+				t.Log("ok")
+			}
+		})
+	}
+}
+
+func TestReceiveFromChannel(t *testing.T) {
+
+	//Since SendToChannel has already been tested, only buffered chan will be considered here
```

Comment:
```suggestion
	// Since SendToChannel has already been tested, only buffered chan will be considered here
```
Please follow this comment formatting standard.